### PR TITLE
4859 Layouts and staticpages

### PR DESCRIPTION
**Pull request by iamanhquan**
https://edu-redmine.framgia.vn/issues/4859

### CODE REVIEW

**Comment by iamanhquan** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 1]
```diff
@@ -1,87 +1,189 @@
...
+
+p {
+  font-size: 1.1em;
+  line-height: 1.7em;
+}
+
+footer {
+  margin-top: 45px;
+  padding-top: 5px;
+  border-top: 1px solid #eaeaea;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Color variable

File: app/assets/stylesheets/custom.scss [hunk 2]
```diff
@@ -1,87 +1,189 @@
...
+body {
+
+  margin: 0;}
+
+body {
+  font-family: -apple-system, system-ui, BlinkMacSystemFont, "Segoe UI", "Helvetica Neue", sans-serif;
+  font-size: 2rem;
+  font-weight: normal;
+  line-height: 1.5;
+  color: #292b2c;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Please use color variable

File: app/assets/stylesheets/custom.scss [hunk 3]
```diff
@@ -1,87 +1,189 @@
...
+}
+
+html {
+  font-family: sans-serif;
+  line-height: 1.15;
+  -ms-text-size-adjust: 100%;
+  -webkit-text-size-adjust: 100%;}
+
+body {
+
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Remove this empty line inside a block

File: app/assets/stylesheets/custom.scss [hunk 4]
```diff
@@ -1,87 +1,189 @@
...
+nav,
+section {
+  display: block;}
+
+h1 {
+  font-size: 2em;
+  margin: 0.67em 0;}
+
+a:hover{
+    text-decoration: none;
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: 2 space indentation

File: app/assets/stylesheets/custom.scss [hunk 5]
```diff
@@ -1,87 +1,189 @@
...
+}
+
+.navbar-nav.navbar-right .btn {z-index: 2; padding: 4px 20px; margin: 10px auto;}
+.navbar .navbar-collapse .navbar-right > li:last-child {padding-left: 22px;}
+
+.navbar .nav-collapse {position: absolute; z-index: 1; top: 0; left: 0; right: 0; bottom: 0; margin: 0; padding-right: 120px; padding-left: 80px; width: 100%;}
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Please use the same style of writing CSS code throughout the file.

File: app/assets/stylesheets/custom.scss [hunk 6]
```diff
@@ -1,87 +1,189 @@
...
+.navbar.navbar-default .nav-collapse {background-color: white;}
+.navbar.navbar-inverse .nav-collapse {background-color: darkslategrey;}
+.navbar .nav-collapse .navbar-form {border-width: 0; box-shadow: none;}
+.nav-collapse>li {float: right;}
+
+.btn.btn-circle {border-radius: 50px;}
+.btn.btn-outline {background-color: transparent;}
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Please correct it for the above lines

File: app/helpers/static_pages_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module StaticPagesHelper
```
Comment (mdmahmudurrahman) [inline app/helpers/static_pages_helper.rb]: Please remove this unused helper and you can add when it is necessary

File: app/views/layouts/_footer.html.erb [hunk 8]
```diff
@@ -0,0 +1,9 @@
+<footer>
+<nav class="navbar navbar-default navbar-fixed-bottom">
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]: Add 2 space indentation as this nav is nested inside the footer tag

File: app/views/layouts/_footer.html.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+  <div class="footer-container">
+    <div class="col-lg-12">
+      <center><p class="cpr"><%= t".copyright" %></p></center>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_footer.html.erb]:
```ruby
<%= t ".copyright" %>
```

File: app/views/layouts/_header.html.erb [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  <div class="container">
+    <div class="navbar-header">
+      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse-2">
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <a href="#">
+        <img width="50" src="/assets/layout/logo.png" border="0"/>
+        <span style="color: orange; font-weight: bold;">FChat - Connected</span>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: 1. I18n
2. Please move the style to an external file

File: app/views/static_pages/contact.html.erb [hunk 11]
```diff
@@ -0,0 +1,8 @@
+<div class="body-container">
+  <br>
+  <iframe src="https://www.google.com/maps/embed?pb=!1m14!1m8!1m3!1d14897.601627517693!2d105.7818972!3d21.0166589!3m2!1i1024!2i768!4f13.1!3m3!1m2!1s0x0%3A0x56c7da1281efdc2f!2sHandico+Tower!5e0!3m2!1svi!2s!4v1546491669632" width="600" height="450" frameborder="0" style="border:0" allowfullscreen></iframe>
+  <div class="col-md-5">
+    <h1 class="title"><%= t".contact" %></h1>
+    <h2 class="subtitle"><%= t".assist" %></h2>
+  </div>
+</div>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/contact.html.erb]: Please add a new line after the end of a file

File: app/views/static_pages/home.html.erb [hunk 12]
```diff
@@ -0,0 +1,14 @@
+<div class="body-container">
+  <div class="col-md-4">
+    <center>
+      <br>
+      <h1><%= t".join" %></h1>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]:
```ruby
<%= t ".join" %>
```

File: app/views/static_pages/home.html.erb [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+      <%= link_to t(".signup"), "#", class: "btn btn-lg btn-primary" %>
+    </center>
+  </div>
+  <div class="col-md-8">
+    <br>
+    <center>
+      <img src="/assets/bg_vision.jpg" width="700px" height="450px"></center>
+  </div>
+</div>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/home.html.erb]: Add new line

File: app/views/static_pages/service.html.erb [hunk 14]
```diff
@@ -0,0 +1,4 @@
+<div class="body-container">	
+  <br>
+  <center><h1><%= t".incoming" %></h1></center>
+</div>
```
Comment (mdmahmudurrahman) [inline app/views/static_pages/service.html.erb]: The same here

File: config/routes.rb [hunk 15]
```diff
@@ -2,5 +2,9 @@
   scope "(:locale)", locale: /en|vi/ do
     get "/login", to: "sessions#new"
     get "/signup", to: "users#new"
+  	get "/about", to: "static_pages#about"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Convention: indentation

**Comment by mdmahmudurrahman** [comment]
commented
Please self review the convention before sending pr. (bow)

File: Gemfile [hunk 16]
```diff
@@ -4,12 +4,19 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
...
 gem "mysql2", ">= 0.4.4", "< 0.6.0"
 gem "turbolinks", "~> 5"
-gem "bootstrap-sass", "3.3.7"
+gem "bootstrap-sass"
 gem "jquery-rails"
 gem "rails-i18n"
+gem "puma", "~> 3.11"
+gem "sass-rails", "~> 5.0"
+gem "uglifier", ">= 1.3.0"
+gem "coffee-rails", "~> 4.2"
```
Comment (huuphuc93) [inline Gemfile]: xóa gem này đi nhé em, dùng js thuần

File: app/assets/stylesheets/custom.scss [hunk 17]
```diff
@@ -1,87 +1,228 @@
...
+  left: 0;
+  right: 0;
+  bottom: 0;
+  margin: 0;
+  padding-right: 120px; padding-left: 80px;
+  width: 100%;
+}
+
+.navbar.navbar-default .nav-collapse {
+  background-color: white;
```
Comment (huuphuc93) [inline app/assets/stylesheets/custom.scss]: dùng biến màu

File: app/assets/stylesheets/custom.scss [hunk 18]
```diff
@@ -1,87 +1,228 @@
...
+  font-weight: normal;
+  line-height: 1.5;
+  color: $light_black;
+  background-image: url(/assets/background.jpeg);
+  background-size: cover;
+  background-attachment: fixed;
+}
+
+  body.has-bg [canvas=container] {
+  background-color: whitesmoke;
```
Comment (huuphuc93) [inline app/assets/stylesheets/custom.scss]: dùng biến màu

File: app/controllers/static_pages_controller.rb [hunk 19]
```diff
@@ -0,0 +1,7 @@
+class StaticPagesController < ApplicationController
+  def home
```
Comment (huuphuc93) [inline app/controllers/static_pages_controller.rb]: def home; end

File: config/routes.rb [hunk 20]
```diff
@@ -2,5 +2,9 @@
   scope "(:locale)", locale: /en|vi/ do
     get "/login", to: "sessions#new"
     get "/signup", to: "users#new"
+    get "/about", to: "static_pages#about"
+    get "/contact", to: "static_pages#contact"
+  	get "/service", to: "static_pages#service"
```
Comment (huuphuc93) [inline config/routes.rb]: indent

**Comment by huuphuc93** [comment]
commented
2 commit giống nhau gộp lại thành 1 nhé em

File: config/routes.rb (hunk 20 above)
Comment (iamanhquan) [reply config/routes.rb]: cái này e fix rồi mà push lên nó vẫn hiện vậy :(((

**Comment by iamanhquan** [comment]
Ready

File: app/assets/stylesheets/custom.scss [hunk 21]
```diff
@@ -1,87 +1,228 @@
...
-  height: 40px;
-}
+@import 'bootstrap-sprockets';
+@import 'bootstrap';
+@import '_color.scss';
+
+.img-logo{
+  display: block;
+  margin: -75px auto 0 auto;
+
```
Comment (mdmahmudurrahman) [inline app/assets/stylesheets/custom.scss]: Remove empty line inside a block

File: app/views/layouts/_header.html.erb [hunk 22]
```diff
@@ -0,0 +1,28 @@
...
+  <div class="container">
+    <div class="navbar-header">
+      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar-collapse-2">
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+        <span class="icon-bar"></span>
+      </button>
+      <a href="#">
+        <img width="50" src="/assets/layout/logo.png" border="0"/>
+        <span class="slogan" "><%= t ".slogan" %></span>
```
Comment (mdmahmudurrahman) [inline app/views/layouts/_header.html.erb]: Remove redundant comma "

File: config/routes.rb (hunk 20 above)
Comment (mdmahmudurrahman) [reply config/routes.rb]: @huuphuc93 I see that, it is not fixed. Please remove extra white space you added and align it with line 4

**Comment by mdmahmudurrahman** [comment]
commented

File: config/routes.rb (hunk 20 above)
Comment (iamanhquan) [reply config/routes.rb]: @mdmahmudurrahman  i fixed that in my local, but when i push here, it don't work

**Comment by iamanhquan** [comment]
Ready

File: config/routes.rb [hunk 23]
```diff
@@ -2,5 +2,9 @@
   scope "(:locale)", locale: /en|vi/ do
     get "/login", to: "sessions#new"
     get "/signup", to: "users#new"
+  	get "/about", to: "static_pages#about"
+  	get "/contact", to: "static_pages#contact"
+    get "/service", to: "static_pages#service"
+  	root "static_pages#home"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: Please do the same for all of the lines. :)))

File: config/routes.rb [hunk 24]
```diff
@@ -2,5 +2,9 @@
   scope "(:locale)", locale: /en|vi/ do
     get "/login", to: "sessions#new"
     get "/signup", to: "users#new"
+  	get "/about", to: "static_pages#about"
+  	get "/contact", to: "static_pages#contact"
```
Comment (mdmahmudurrahman) [inline config/routes.rb]: The same here!

**Comment by mdmahmudurrahman** [comment]
commented

**Comment by iamanhquan** [comment]
Ready

**Comment by mdmahmudurrahman** [comment]
merged
CC: [To:1818111] Nguyen Thi Huong (94)

**Pull request merged**